### PR TITLE
mediatek: filogic: add support for Edgecore EAP111

### PR DESCRIPTION
**Pull request by robimarko**
HW specifications:
* Mediatek MT7981A
* 256MB SPI-NAND
* 512MB DRAM
* Uplink: 1 x 10/100/1000Base-T Ethernet, Auto MDIX, RJ-45 with 802.3at
PoE (Built-in GBe PHY)
* LAN: 1 x 10/100/1000Base-T Ethernet, Auto MDIX, RJ-45 (Airoha EN8801SC)
* 1 Tricolor LED
* Reset button
* 12V/2.0A DC input

Installation:
Board comes with OpenWifi/TIP which is OpenWrt based, so sysupgrade can
be used directly over SSH.

Notes:
Unfortunately, there is no specification sheet nor datasheet available
in order to demistify the magic PBUS writes and work on upstreaming
the Airoha EN8801SC PHY driver.

Signed-off-by: Robert Marko <robert.marko@sartura.hr>

### CODE REVIEW

File: target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+				reg = <0x00000 0x100000>;
+				read-only;
+			};
+
+			partition@100000 {
+				label = "u-boot-env";
+				reg = <0x100000 0x80000>;
+			};
+
+			factory: partition@180000 {
```
Comment (musashino205) [inline target/linux/mediatek/dts/mt7981a-edgecore-eap111.dts:121]: ditto

**Comment by aiamadeus** [comment]
```
#define EN8801S_PHY_ID1         0x03a2
#define EN8801S_PHY_ID2         0x9461
#define EN8801SC_PHY_ID         0x03a29471
```
https://github.com/torvalds/linux/tree/master/drivers/net/phy/mediatek-ge-soc.c#11:
```
#define MTK_GPHY_ID_MT7981			0x03a29461
```
It seems that MT7981 SoC has a built-in EN8801S PHY. Maybe EN8801SC
can be supported by the MEDIATEK_GE_SOC_PHY driver?

**Comment by robimarko** [comment]
@aiamadeus Well, that is the thing, there isn't anything public about this PHY, quite literally not even a product brief or anything, the MTK driver is the only resource and its just a bunch of magic writes.

**Comment by musashino205** [comment]
from CI log:

```
    ERROR: target/linux failed to build.
make -r target/compile: build failed. Please re-run make with -j1 V=s or V=sc for a higher verbosity level to see what's going on
make: *** [/__w/openwrt/openwrt/openwrt/include/toplevel.mk:233: target/compile] Error 1
====== Make errors from logs/target/linux/compile.txt ======
 1034 |         struct device *dev = &mbus->dev;
      |                        ^~~
drivers/net/phy/en8801sc.c: At top level:
drivers/net/phy/en8801sc.c:102:12: error: 'airoha_cl22_write' defined but not used [-Werror=unused-function]
  102 | static int airoha_cl22_write(struct mii_bus *ebus, int phy_addr,
      |            ^~~~~~~~~~~~~~~~~
drivers/net/phy/en8801sc.c:95:21: error: 'airoha_cl22_read' defined but not used [-Werror=unused-function]
   95 | static unsigned int airoha_cl22_read(struct mii_bus *ebus, int phy_addr,
      |                     ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[9]: *** [scripts/Makefile.build:243: drivers/net/phy/en8801sc.o] Error 1
```

**Comment by castiel652** [comment]
@robimarko I think you can try sending an email to Airoha.

Someone on the BPI forum manged to got something from them for the EN8811H PHY.

**Comment by robimarko** [comment]
@musashino205 Thanks, Werror locally now flushed all warnings turned errors.

@castiel652 Something like that is already in motion, but it will take time even if successful, in the mean time I see one more device also using the same PHY

**Comment by robimarko** [comment]
Thanks! Rebased on top of main and merged!

**Pull request merged**